### PR TITLE
Forward refs for Tagline

### PR DESCRIPTION
**Pull request by nishasy**
## Summary:
We want to forward refs for Tagline. To make this easier,
I converted it to a functional component.

I also changed the test file that I had created for Title
so that it accomodates multiple elements instead. This will
make it much easier going forward for the other typography
elements.

Issue: https://khanacademy.atlassian.net/browse/WB-424

## Test plan:
`yarn jest consistency-tests/__tests__/ref-forwarded.test.tsx`

Check that local storybook typography page matches the prod storybook typography page.

### CODE REVIEW

File: packages/wonder-blocks-typography/src/components/__tests__/typography.test.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import Title from "../title";
+import Tagline from "../tagline";
+
+describe("Typography elements", () => {
+    test.each`
+        Component  | name         | type                  | typeName
+        ${Tagline} | ${"Tagline"} | ${HTMLSpanElement}    | ${"HTMLSpanElement"}
+        ${Title}   | ${"Title"}   | ${HTMLHeadingElement} | ${"HTMLHeadingElement"}
+    `("$name forwards ref to $typeName", async ({Component, type}: any) => {
```
Comment (kevinbarabash) [inline packages/wonder-blocks-typography/src/components/__tests__/typography.test.tsx:12]: Nice!

File: packages/wonder-blocks-typography/src/components/__tests__/typography.test.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        // Arrange
+        const ref: React.RefObject<HTMLHeadingElement> = React.createRef();
+
+        // Act
+        render(<Component ref={ref}>This is a title</Component>);
+
+        // Assert
+        await waitFor(() => {
+            expect(ref.current).toBeInstanceOf(type);
+        });
```
Comment (kevinbarabash) [inline packages/wonder-blocks-typography/src/components/__tests__/typography.test.tsx:22]: Is `waitFor` necessary here?

**Review by kevinbarabash — APPROVED** [review]
LGTM 🎉

**Comment by nishasy** [comment]
The parent pull-request ([#1941](https://github.com/Khan/wonder-blocks/pull/1941)) has been merged into `main`, but this branch (`add-forwarded-refs-to-tagline`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

**Pull request merged**